plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-block/additional-checkout-fields.md

```diff
@@ -0,0 +1,388 @@
+| `id`            | The field's ID. This should be a unique identifier for your field. It is composed of a namespace and field name separated by a `/`.    | Yes       | `plugin-namespace/how-did-you-hear`          | No default - this must be provided.                                        |
+| `label`         | The label shown on your field. This will be the placeholder too.                                                                       | Yes       | `How did you hear about us?`                 | No default - this must be provided.                                        |
+| `optionalLabel` | The label shown on your field if it is optional. This will be the placeholder too.                                                     | No        | `How did you hear about us? (Optional)`      | The default value will be the value of `label` with `(optional)` appended. |
+| `location`      | The location to render your field.                                                                                                     | Yes       | `contact`, `address`, or `additional`        | No default - this must be provided.                                        |
+| `type`          | The type of field you're rendering. It defaults to `text` and must match one of the supported field types.                             | No        | `text`, `select`, or `checkbox`              | `text`                                                                     |
+| `attributes`    | An array of additional attributes to render on the field's input element. This is _not_ supported for `select` fields.                 | No        | `[	'data-custom-data' => 'my-custom-data' ]` | `[]`                                                                       |
+
+#### Options for `text` fields
+
+As well as the options above, text fields also support a `required` option. If this is `true` then the shopper _must_ provide a value for this field during the checkout process.
+
+| Option name     | Description                                                                                                                         | Required? | Example                                      | Default value |
+|-----------------|-------------------------------------------------------------------------------------------------------------------------------------|-----------|----------------------------------------------|---|
+| `required` | If this is `true` then the shopper _must_ provide a value for this field during the checkout process. | No | `true` | `false` |
+
+#### Options for `select` fields
+
+As well as the options above, select fields must also be registered with an  `options` option. This is used to specify what options the shopper can select.
+
+The `optionalLabel` option will never be shown as select fields are _always_ required.
```

Comment:
I added a line to the select form, but the checkbox one already reads:

> The checkbox field type does not have any specific options, however `required` will always be `false` for a checkbox field. Making a checkbox field required is not supported.

Not sure how much clearer this can be? Do you have a suggestion to reword that?

Reply:
Just asking why required is top level like this and not input attributes? Could required at server level be handled by validation_callbacks instead?